src/index.js

```diff
@@ -20,10 +20,185 @@
+        'managerHostPort': {'type': 'string'}
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let jaegerSchema = {
+    'id': '/jaeger',
+    'type': 'object',
+    'required': ['serviceName'],
+    'properties': {
+        'serviceName': {'type': 'string'},
+        'disable': {'type': 'boolean'},
+        'sampler': {
+            'type': 'object',
+            // should be 'oneOf', but that doesn't work properly with this implementation.
+            'anyOf': [
+                {'$ref': 'const'},
+                {'$ref': 'probabilistic'},
+                {'$ref': 'ratelimiting'},
+                {'$ref': 'remote'}
+            ]
+        },
+        'reporter': {
+            'properties': {
+                'logSpans': {'type': 'boolean'},
+                'agentHost': {'type': 'string'},
+                'agentPort': {'type': 'number'},
+                'flushIntervalMs': {'type': 'number'}
+            },
+            'additionalProperties': false
+        }
+    }
+};
+
+let getSampler = (options) => {
+    let type = options.sampler.type;
+    let param = options.sampler.param;
+
+    let sampler;
+    if (type === 'probabilistic') {
+       sampler = new ProbabilisticSampler(param);
+    }
+
+    if (type === 'ratelimiting') {
+        sampler = new RatelimitingSampler(param);
+    }
+
+    if (type === 'const') {
+        sampler = new ConstSampler(param);
+    }
+
+    if (type === 'remote') {
+        sampler = new RemoteSampler(options.serviceName);
+    }
+
+    return sampler;
+}
+
+let initTracer = (options) => {
+    let v = new Validator();
+    v.addSchema(constSamplerSchema);
+    v.addSchema(probabilisticSamplerSchema);
+    v.addSchema(ratelimitingSamplerSchema);
+    v.addSchema(remoteSamplerSchema);
+    v.addSchema(jaegerSchema);
+    v.validate(options, configSchema, {
+        throwError: true
+    });
+
+    options = options.jaeger;
+
+    let reporter = null;
+    let sampler = null;
+    if (options.disable) {
+        reporter = new NoopReporter();
+        sampler = new ConstSampler(false);
+    } else {
+        let sender;
+        let reporterOptions = {};
+        if (options.reporter) {
+            if (options.reporter.logSpans) {
+                // use logging reporter
+            }
+            if (options.reporter.flushIntervalMs) {
+                reporterOptions['bufferFlushInterval'] = options.reporter.flushIntervalMs;
+            }
+
+            if (options.reporter.agentHost && options.reporter.agentPort) {
+                let hostPort = `${options.reporter.agentHost}:${options.reporter.agentPort}`;
+                sender = new UDPSender(hostPort);
+            } else {
+                // use default sender
+                sender = new UDPSender();
+            }
+        } else {
+            // use default sender
+            sender = new UDPSender();
+        }
+
+
+        reporter = new RemoteReporter(sender, reporterOptions);
```

Comment:
we can use the same pattern as with sampler - if `options.reporter` is defined, use it, otherwise create remote. This is helpful when people are testing and *only* want a logging reporter. This came up in Java lib once, where there's no way to override the reporter via config.